Media/Audio/OpenALSoundProvider.cpp

```diff
@@ -118,35 +118,48 @@ void OpenALSoundProvider::Release() {
     }
 }
 
+void OpenALSoundProvider::DeleteBuffers(StreamingTrackBuffer *track, int type) {
+    int count = 0;
+    alGetSourcei(track->source_id, type, &count);
+    if (CheckError()) {
+        log("OpenAL: Fail to get buffers count.");
+        assert(false);
+    }
+    if (count > 0) {
+        unsigned int *buffer_ids = new unsigned int[count];
+        alSourceUnqueueBuffers(track->source_id, count, buffer_ids);
```

Comment:
Hi, this line plagues the game with assertions now as alSourceUnqueueBuffers returns AL_INVALID_VALUE when you call DeleteBuffers with AL_BUFFERS_QUEUED
The reason for that I assume is that (according to https://www.openal.org/documentation/OpenAL_Programmers_Guide.pdf )  , alSourceUnqueueBuffers expects some valid buffer ids in buffer_ids, but that array is never initialized

Sound works alright in principle, but every time any sound finishes playing this code throws an assertion exception

Reply:
I can't find universal way to fix this for windows and macOS in same time, and only way what i found is branching critical code.